x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/action/GetDataFrameJobsStatsAction.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ml.featureindexbuilder.action;
+
+import org.elasticsearch.action.Action;
+import org.elasticsearch.action.ActionRequestBuilder;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.FailedNodeException;
+import org.elasticsearch.action.TaskOperationFailure;
+import org.elasticsearch.action.support.tasks.BaseTasksRequest;
+import org.elasticsearch.action.support.tasks.BaseTasksResponse;
+import org.elasticsearch.client.ElasticsearchClient;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.xpack.ml.featureindexbuilder.job.FeatureIndexBuilderJob;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class GetDataFrameJobsStatsAction extends Action<GetDataFrameJobsStatsAction.Response> {
```

Comment:
For anomaly detection, the equivalent action is called `GetJobStatsAction` where job is singular. We might want to be consistent but I don't feel strongly about it.

Reply:
Discussed this with @droberts195 on a previous PR and we went all for plural, all the endpoint have changed to `.../jobs/{id}/..`. `GetDataFrameJob` changed to `GetDataFrameJobs`, etc.

Anyway, it's a good point and I suggest to keep it as is for now but have a session about naming where we can go over all endpoints and then do a renaming PR.

Reply:
The ML actions are inconsistent in this respect: we have `GetJobsStatsAction` and `GetDatafeedsStatsAction` but `GetJobStatsActionRequest` and `RestGetJobStatsAction`.  I think plural is correct.